ospfd/ospf_auth.c

```diff
@@ -159,6 +159,8 @@ static int ospf_auth_check_md5_digest(struct ospf_interface *oi,
 	struct crypt_key *ck = NULL;
 	uint16_t length = ntohs(ospfh->length);
 
+	assert(length <= UINT16_MAX); /* for coverity's sake */
```

Comment:
whoa - we can't blow up when looking at an incoming packet, can we? can't we just do a test and reject an invalid value?

Reply:
Well, we haven't blown up, and won't. But for some reason, coverity thinks we may be wrong.  
We can change the `UINT16_MAX` value with `sizeof(struct ospf_header) + KEYCHAIN_MAX_HASH_SIZE` for a better test.

Reply:
no, really: please don't assert on incoming packets, please construct a reasonable test and avoid processing if the test fails. surely that will satisfy coverity too?

Reply:
You are right. I changed the assertion test right after your comment. Please check the new assertion.

Reply:
no, really: please don't assert. if there's a test that can fail, then just invalidate, drop, etc. the packet.

Reply:
Done